nixos/modules/hardware/tuxedo-keyboard.nix

```
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ... }:
+
+# TODO: look at the other hardware modules
+# TODO: re-build for the specific kernel
+# TODO: I am not quite sure IF a module is still needed here since the configurations here could be set directly without this module. 
+# 
+#         boot.kernelParams = [
+#           "tuxedo_keyboard.mode=0"
+#           "tuxedo_keyboard.brightness=${toString cfg.brightness}"
+#           "tuxedo_keyboard.color_left=0x${cfg.color}"
+#         ]; 
+with lib;
+
+let 
+  # TODO: load tuxedo-keyboard from the kernel Packages
+  cfg = config.tuxedo-keyboard;
+  tuxedo-keyboard = config.boot.kernelPackages.tuxedo-keyboard;
+in
+  {
+    options.tuxedo-keyboard = {
+      enable = mkEnableOption "tuxedo-keyboard";
+    };
+
+    config = mkIf cfg.enable 
+    {
+         boot.kernelModules = ["tuxedo_keyboard"];
+         boot.extraModulePackages = [ tuxedo-keyboard ];
+         environment.systemPakcages = [ pkgs.tuxedo-keyboard ];
```

```suggestion
         environment.systemPackages = [ tuxedo-keyboard ];
```
Is that required? Is there an accompanying CLI for it?

---

Oh no there isn't. That was a mistake :) Thanks for catching that! :)

---

removed it altoether :)